app/services/code_owners_service.rb

```diff
@@ -0,0 +1,30 @@
+class CodeOwnersService < BaseService
+  def call
+    Project.find_each do |project|
+      content_file = GithubReposApi.new(project.name).get_content_from_file('CODEOWNERS')
+      next if content_file.empty?
+
+      code_owners_names = build_code_owners_array(content_file)
+      remove_old_code_owners(code_owners_names, project)
+      add_new_code_owners(code_owners_names, project)
+    end
+  end
+
+  def add_new_code_owners(code_owners_names, project)
+    current_code_owners = project.code_owners
+    code_owners_names.each do |code_owner_name|
+      user = User.find_by(login: code_owner_name)
```

Comment:
What if the user hasn't been created yet? Is it possible? Perhaps `find_or_create_by` will be better

Reply:
in that flow we do not have all the attributes that are needed to create a user, in theory the cases where the user doesn't exist yet should be few..  we could do a `next` to skip it for the moment until a PR, Review or Review Request is created. what do you think?